File: SphrLibrary/Entities/OpenmHealth/ValueUnit.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth ValueUnit リソースを表します。
    /// </summary>
    [DataContract()]
    [Serializable()]
    public class ValueUnit
    {
        [DataMember()]
        public double value = 0;

        [DataMember()]
        public string unit = string.Empty;

        public ValueUnit() { }
    }
}

File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/ValueUnit.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

/**
 * Open mHealth での 測定値と単位
 */
@Data
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class ValueUnit {

    /**
     * 測定値
     */
    private double value = 0.0;

    /**
     * 単位
     */
    @NonNull
    private String unit = "";
}
